txpool/txpool.go

```diff
@@ -724,90 +739,111 @@ func (p *TxPool) pruneAccountsWithNonceHoles() {
 // (only once) and an enqueueRequest is signaled.
 func (p *TxPool) addTx(origin txOrigin, tx *types.Transaction) error {
 	if p.logger.IsDebug() {
-		p.logger.Debug("add tx",
-			"origin", origin.String(),
-			"hash", tx.Hash.String(),
-		)
+		p.logger.Debug("add tx", "origin", origin.String(), "hash", tx.Hash.String())
 	}
 
 	// validate incoming tx
 	if err := p.validateTx(tx); err != nil {
 		return err
 	}
 
+	// calculate tx hash
+	tx.ComputeHash()
+
+	// initialize account for this address once or retrieve existing one
+	account := p.getOrCreateAccount(tx.From)
+	// populate currently free slots
+	slotsFree := p.gauge.max - p.gauge.read()
```

Comment:
Maybe create a function on gauge, and call it `freeSlots`?

Reply:
fixed https://github.com/0xPolygon/polygon-edge/pull/1656/commits/bbb28b28bd5e4766631fd6d00211e59610dc77c1